--- a/dlls/winex11.drv/fs.c
+++ b/dlls/winex11.drv/fs.c
@@ -86,12 +86,16 @@ static struct fs_monitor_size fs_monitor
     {1920, 1280}, /*  3:2 */
     {1920, 800},  /* 12:5 */
     {3840, 1600}, /* 12:5 */
-    {1720, 720}, /* 21:9 - 'FSR ultra-wide Performance' */
-    {2024, 847}, /* 21:9 - 'FSR ultra-wide Balanced' */
-    {2293, 960}, /* 21:9 - 'FSR ultra-wide Quality' */
-    {2560, 1080}, /* 21:9 ultra-wide */
-    {2646, 1108}, /* 21:9 - 'FSR ultra-wide Ultra Quality' */
-    {3440, 1440}, /* 21:9 ultra-wide */
+    {1280, 540},  /* 21:9 - 'FSR ultrawide 1080p Performance' */
+    {1506, 635},  /* 21:9 - 'FSR ultrawide 1080p Balanced' */
+    {1706, 720},  /* 21:9 - 'FSR ultrawide 1080p Quality' */
+    {1970, 831},  /* 21:9 - 'FSR ultrawide 1080p Ultra Quality' */
+    {2560, 1080}, /* 21:9 ultrawide 1080p */
+    {1720, 720},  /* 21:9 - 'FSR ultrawide 1440p Performance' */
+    {2024, 847},  /* 21:9 - 'FSR ultrawide 1440p Balanced' */
+    {2293, 960},  /* 21:9 - 'FSR ultrawide 1440p Quality' */
+    {2646, 1108}, /* 21:9 - 'FSR ultrawide 1440p Ultra Quality' */
+    {3440, 1440}, /* 21:9 ultrawide 1440p */
     {1280, 1024}, /*  5:4 */
 };
 
