--- a/dlls/kernelbase/memory.c
+++ b/dlls/kernelbase/memory.c
@@ -107,8 +107,8 @@ void WINAPI DECLSPEC_HOTPATCH GetSystemI
     si->dwPageSize                  = basic_info.PageSize;
     si->lpMinimumApplicationAddress = basic_info.LowestUserAddress;
     si->lpMaximumApplicationAddress = basic_info.HighestUserAddress;
-    si->dwActiveProcessorMask       = basic_info.ActiveProcessorsAffinityMask;
-    si->dwNumberOfProcessors        = basic_info.NumberOfProcessors;
+    si->dwActiveProcessorMask       = 255; /*basic_info.ActiveProcessorsAffinityMask;*/
+    si->dwNumberOfProcessors        = 8;   /*basic_info.NumberOfProcessors;*/
     si->dwAllocationGranularity     = basic_info.AllocationGranularity;
     si->wProcessorLevel             = cpu_info.Level;
     si->wProcessorRevision          = cpu_info.Revision;
